src/pages/Missions/Missions.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchMissionAsync, joiningMission, leavingMission } from '../../redux/missionSlice';
+import './Missions.css';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions.missions);
+  const status = useSelector((state) => state.missions.status);
+  const error = useSelector((state) => state.missions.error);
+
+  useEffect(() => {
+    if (missions.length === 0) {
+      dispatch(fetchMissionAsync());
+    }
+  }, [dispatch]);
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'failed') {
+    return (
+      <div>
+        Error:
+        {error}
+      </div>
+    );
+  }
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>Mission</th>
+          <th>Description</th>
+          <th>Status</th>
+          <th> </th>
+        </tr>
+      </thead>
+      <tbody className="border border-gray-300">
+        {missions.map((mission) => (
+          <tr key={mission.mission_id}>
+            <td>{mission.mission_name}</td>
+            <td>{mission.description}</td>
+            <td>
+              <p
+                style={{
+                  backgroundColor: mission.reserved ? '#419bf9' : '#6d757d',
+                }}
+              >
+                {mission.reserved ? 'ACTIVE MEMBER' : 'NOT A MEMBER'}
+              </p>
+            </td>
+            <td>
+              {mission.reserved && (
+              <button
+                type="button"
+                style={{
+                  color: mission.reserved ? 'red' : '',
+                  border: mission.reserved ? '1px solid red' : '',
+                }}
+                onClick={() => dispatch(leavingMission(mission.mission_id))}
+              >
+                Leave Mission
+              </button>
+              )}
+              {!mission.reserved && (
+              <button
+                type="button"
+                style={{
+                  color: mission.reserved ? 'red' : '',
+                  border: mission.reserved ? '1px solid red' : '',
+                }}
+                onClick={() => dispatch(joiningMission(mission.mission_id))}
+              >
+                Join Mission
+              </button>
```

Comment:
👍 Great job on your implementation! However, there's a slight issue with the button display:

- [ ] In the design, the button is displayed as a block-level element, but in your code, it appears to be broken into multiple lines. To align with the design, please update your button element to be displayed as a block element.
- [ ] Second, please also update the button background color to match the design as much as possible.

| Your | Design |
| --- | --- |
|  ![image](https://github.com/pushkar1995/space-travelers-hub/assets/55840999/86dd5b3d-4cc3-4264-ade4-5a1ec56cb547)| ![image](https://github.com/pushkar1995/space-travelers-hub/assets/55840999/11de4b30-eb30-4f64-9870-9ee0ee82b7e8)|

By making this adjustment, you'll ensure that your button layout matches the design specifications. Keep up the excellent work! 👍🎨